### PR TITLE
Changing scalability instance from spot to onDemand

### PR DESCRIPTION
**Pull request by meghna-khemka**
# Description

Changing scalability instance from spot to onDemand

medic/cht-core#[number]

# Code review checklist

- [x ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/scalability/start_ec2_medic.sh [hunk 1]
```diff
@@ -1,20 +1,11 @@
...
-echo Getting Instance ID
-instanceID=$(aws ec2 describe-spot-instance-requests --spot-instance-request-ids $SpotInstanceRequestId | jq .SpotInstanceRequests[0].InstanceId -r )
-while [ "$instanceID" = null ]
-do
-sleep 5
-instanceID=$(aws ec2 describe-spot-instance-requests --spot-instance-request-ids $SpotInstanceRequestId | jq .SpotInstanceRequests[0].InstanceId -r )
-echo "Sleeping while waiting for instance ID to be setup."
+
+echo Triggering EC2 Run Instance Command and getting Instance ID
+instanceID=$(aws ec2 run-instances --image-id ami-065ba2b6b298ed80f --instance-type c5.2xlarge --security-group-ids sg-0fa20cd785acec256 --block-device-mappings file://block-device-mapping.json --user-data file://medic-os.sh --instance-initiated-shutdown-behavior terminate | jq .Instances[0].InstanceId -r )
```
Comment (newtewt) [inline tests/scalability/start_ec2_medic.sh:6]: Nice not having to wait!

File: tests/scalability/medic-os.sh [hunk 2]
```diff
@@ -8,4 +8,5 @@ cd cht-infrastructure/self-hosting/prepare-system/ubuntu/;
 mkdir -p /home/medic/self-hosting/main/
 echo DOCKER_COUCHDB_ADMIN_PASSWORD=medicScalability >> /home/medic/self-hosting/.env
 echo HA_PASSWORD=medicScalability >> /home/medic/self-hosting/.env
-./prepare.sh
\ No newline at end of file
+./prepare.sh
+sudo shutdown -P +60
```
Comment (newtewt) [inline tests/scalability/medic-os.sh:12]: Move the shutdown to the beginning of the script. That way if something fails before hand the shutdown is triggered and the ec2 instance won't just hang around. I think 60 minutes should be fine for now and if we run into issues we can bump it up.

File: tests/scalability/run_suite.sh [hunk 3]
```diff
@@ -31,10 +31,11 @@ echo "jmeter do it!"
...
+sudo apt-get install unzip -y
 curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
 unzip awscliv2.zip
 sudo ./aws/install
 echo "Uploading logs and screenshots to ${S3_PATH}..."
 /usr/local/bin/aws s3 cp ./report "$S3_PATH" --recursive
-echo "FINISHED! "
\ No newline at end of file
+echo "FINISHED! "
+sudo shutdown -P +60
```
Comment (newtewt) [inline tests/scalability/run_suite.sh:41]: Same here. The shutdown can be queued first.

File: tests/scalability/start_jmeter_ec2.sh [hunk 4]
```diff
@@ -8,7 +8,7 @@ sed -i '2s~^~'MEDIC_URL=$MEDIC_URL'\n~' run_suite.sh
...
 sed -i '2s~^~'TAG_NAME=$TAG_NAME'\n~' run_suite.sh
 cat run_suite.sh
-# base64 encode our script and set as value for UserDAta in launch-specification
-jq --arg arn $SCALABILITY_ARN '.UserData = "'$(base64 run_suite.sh -w 0)'" | .IamInstanceProfile.Arn = $arn' launch-specification.json >> launch-specification-jmeter.json
-cat launch-specification-jmeter.json
-echo $(aws ec2 request-spot-instances --spot-price '0.333' --instance-count 1 --type 'one-time' --launch-specification file://launch-specification-jmeter.json --block-duration-minutes 60)
\ No newline at end of file
+
+echo Triggering EC2 Run Instance Command and getting Instance ID
+instanceID=$(aws ec2 run-instances --image-id ami-065ba2b6b298ed80f --instance-type c5.2xlarge --security-group-ids sg-0fa20cd785acec256 --block-device-mappings file://block-device-mapping.json --user-data file://run_suite.sh --instance-initiated-shutdown-behavior terminate | jq .Instances[0].InstanceId -r )
```
Comment (newtewt) [inline tests/scalability/start_jmeter_ec2.sh:13]: The $SCALABILITY_ARN is set to be able to upload the jmeter results to S3. It looks like that is removed from here. We'll need that added back so we can review the results in S3. Once the instance is down it'll be destroyed and the results won't be around anymore right?

**Review by newtewt — CHANGES_REQUESTED** [review]
Very nice! Some minor changes but overall looks awesome.

File: tests/scalability/medic-os.sh (hunk 2 above)
Comment (meghna-khemka) [reply tests/scalability/medic-os.sh]: Done

File: tests/scalability/run_suite.sh (hunk 3 above)
Comment (meghna-khemka) [reply tests/scalability/run_suite.sh]: Done

File: tests/scalability/start_jmeter_ec2.sh (hunk 4 above)
Comment (meghna-khemka) [reply tests/scalability/start_jmeter_ec2.sh]: Yea added the role to instance

**Comment by meghna-khemka** [comment]
Hi @newtewt 
I have made the requested changes .Please review.

**Comment by newtewt** [comment]
@meghna-khemka  nice work. I'm approving but looks like there is some build failures.

**Review by newtewt — APPROVED** [review]

**Pull request merged**